spec/models/vm_or_template/operations/configuration_spec.rb

```diff
@@ -14,21 +14,13 @@
 
     context "when ext_management_system exists" do
       let(:vm) { FactoryBot.create(:vm_or_template, :ext_management_system => ems) }
-      let(:ems) { FactoryBot.create(:ext_management_system) }
+      let(:ems) { FactoryBot.create(:ext_management_system, :with_authentication) }
       let(:storage_name) { "test_storage" }
       let(:storage) { FactoryBot.create(:storage, :name => storage_name) }
-      let(:storages) { double("storages") }
-      let(:host) { double("host", :writable_storages => storages) }
-      let(:hosts) { [host] }
-
-      before do
-        allow(ems).to receive(:hosts).and_return(hosts)
-      end
+      let!(:host) { FactoryBot.create(:host, :ext_management_system => ems).tap { |h| h.host_storages.create!(:storage => storage) } }
```

Comment:
Can this possibly be split into two different `let!` statements?  (long line).

Reply:
I didn't need access to the host_storage record by name so I didn't put it in its own let, if anything I would split it out from a one-line `{}` to a multi-line `do;end`